### DISCUSSION TITLE
General queries

### DISCUSSION THREAD

**Discussion by ghost** [Kunzisoft/KeePassDX#1902] [General]
First of all i would like to thank the developer(s) a lot for such an amazing open source and ad-free app. It is the best password manager out there in my opinion. 
   Now coming to the questions, i would like to know:
1- What type of "AES" does this app use in KDF and Encryption?
2- Why does the "merge" option appear (alongwith save,reload option) at many places in the app (such as : pressing 3 dot menu after unlocking db, hamburger menu button>settings>all three options below "database" section) whereas the way to merge databases is  by pressing hamburger menu botton after unlocking the database and then clicking on "merge" and selecting the other database? The "Merge" at other places doesn't show pop-up to add and merge another database. It just appears as to be refreshing the database.
3-Significance/Use of "database description" option (settings> db settings)? I couldn't find its effect anywhere?

**Comment by J-Jamet** [comment]
1 - Uses the same AES encryption in KDF as the standardized KeePass format. https://keepass.info/help/base/security.html
2 - The merge function may differ depending on the file manager you're using. It's most useful when the file URI doesn't change but the file is updated externally, which means that the version in RAM is potentially different from the version that was saved externally while the database was open. In the implementation of many file managers, an external change in the file implies a cache or URI change, which invalidates the link, but I intend to make a manager with much better behavior. https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync#save-merge-and-reload
 3 - Only a metadata describing your use of the database. This is completely optional, perhaps used if you have many different databases.
